gobblin-runtime/src/main/java/org/apache/gobblin/runtime/Task.java

```diff
@@ -394,15 +391,8 @@ public void run() {
     } catch (Throwable t) {
       failTask(t);
     } finally {
-      synchronized (this) {
-        if (this.taskFuture == null || !this.taskFuture.isCancelled()) {
-          this.taskStateTracker.onTaskRunCompletion(this);
-          completeShutdown();
-          this.taskFuture = null;
-        } else {
-          LOG.info("will not decrease count down latch as this task is cancelled");
-        }
-      }
+      this.taskStateTracker.onTaskRunCompletion(this);
```

Comment:
Is the removing of the synchronized block intentional? Is there a chance that Task#cancel() is invoked from multiple threads?